pkg/apis/walruscore/v1/connector_binding.go

```diff
@@ -0,0 +1,55 @@
+package v1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const ProviderLabelPrefix = "walrus.seal.io/provider-"
+
+// ConnectorBinding is the schema for the connectorbindings API.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:crd-gen:resource:scope="Namespaced",subResources=["status"]
+type ConnectorBinding struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ConnectorBindingSpec   `json:"spec,omitempty"`
+	Status ConnectorBindingStatus `json:"status,omitempty"`
+}
+
+var _ runtime.Object = (*ConnectorBinding)(nil)
+
+// ConnectorBindingSpec defines the desired state of ConnectorBinding.
+type ConnectorBindingSpec struct {
+	// ConnectorRef is the reference to the connector.
+	ConnectorRef ConnectorReference `json:"connector_ref"`
+
+	// ConnectorType is the type of the connector.
+	ConnectorType string `json:"connector_type"`
```

Comment:
Why do we need the type

Reply:
Creating connector_binding needs validation for type uniqueness under the same environment. Without the type will need extra queries for each connector.

Reply:
It's determined by the ref, should we put it in status